### DISCUSSION TITLE
Question: Exporting characters as Texture Atlases the proper way?

### DISCUSSION THREAD

**Discussion by AbnormalPoof** [FunkinCrew/Funkin#2855] [Q&A] [answered]
The general consensus is to put every animation in one symbol, so I did that and tried to implement a scuffed Darnell in the game...

... He looks like this in-game.
![Screenshot 2024-06-19 at 3 04 21 AM](https://github.com/FunkinCrew/Funkin/assets/86753001/34e44e28-04e7-447c-9979-13df922330b6)
![Screenshot 2024-06-19 at 3 05 37 AM](https://github.com/FunkinCrew/Funkin/assets/86753001/d3309af4-3f58-44d5-ad2f-a1aaf3bef451)

**Comment by AbnormalPoof** [comment]
UPDATE: I used frame labels, and they kind of work... if only it didn't iterate through literally every single animation in the symbol instead of playing the animations properly.

**Reply by AbnormalPoof** [reply]
Now that I think about it, this might be a Funkin' bug. I also had to set extreme camera offsets just to get him on screen.

**Reply by biomseed** [reply]
> Now that I think about it, this might be a Funkin' bug. I also had to set extreme camera offsets just to get him on screen.

It's not a funkin' bug, it just happened to PLAY the symbol that you put all the frames in, I think. There's some way that funkin characters do it and you just need to figure that out. Maybe the FLAs have an answer, but some of them were made a really long time ago and the ones uploaded to github aren't necessarily always the ones that were used to export the animations.

**Comment by AbnormalPoof** [comment, marked as answer]
I have figured it out.
![Screenshot 2024-08-14 at 1 09 17 AM](https://github.com/user-attachments/assets/e9b0bee9-50c4-4f91-b561-7f3158c57b4c)
For anyone in the future looking for a way to figure out how, put all the frame labels on a **separate layer** instead of labeling the symbol keyframes themselves! This makes it not "iterate" (or play all the symbols in order) and instead plays them properly.

You still have to set extreme camera offsets for now, though.

**Reply by biomseed** [reply]
what's the third layer for? can multiple layers be used?

**Reply by AbnormalPoof** [reply]
For additional frame labels, I suppose. I noticed that Blazin' Darnell does the same thing.
